src/main/java/uk/sky/cirrus/CqlMigratorImpl.java

```diff
@@ -0,0 +1,157 @@
+package uk.sky.cirrus;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.ConsistencyLevel;
+import com.datastax.driver.core.QueryOptions;
+import com.datastax.driver.core.Session;
+import com.google.common.base.Preconditions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.sky.cirrus.exception.ClusterUnhealthyException;
+import uk.sky.cirrus.locking.Lock;
+import uk.sky.cirrus.locking.LockConfig;
+import uk.sky.cirrus.locking.LockService;
+import uk.sky.cirrus.locking.exception.CannotAcquireLockException;
+import uk.sky.cirrus.locking.exception.CannotReleaseLockException;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+/**
+ * Utility class which helps to apply schema changes required by an application prior to application startup.
+ * Makes use of a locking mechanism to ensure only one instance at a time can apply schema changes to the cassandra
+ * database.
+ *
+ */
+
+public final class CqlMigratorImpl implements CqlMigrator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CqlMigratorImpl.class);
+
+    private final LockConfig lockConfig;
+
+    public CqlMigratorImpl(LockConfig lockConfig) {
+        this.lockConfig = lockConfig;
+    }
+
+    /**
+     * Allows cql migrate to be run on the command line.
+     * Set hosts, keyspace and directories using system properties.
+     *
+     * @param args
+     */
+    public static void main(String[] args) {
+
+        final String hosts = System.getProperty("hosts");
+        final String keyspace = System.getProperty("keyspace");
+        final String directoriesProperty = System.getProperty("directories");
+        final String port = System.getProperty("port");
+
+        Preconditions.checkNotNull(hosts, "'hosts' property should be provided having value of a comma separated list of cassandra hosts");
+        Preconditions.checkNotNull(keyspace, "'keyspace' property should be provided having value of the cassandra keyspace");
+        Preconditions.checkNotNull(directoriesProperty, "'directories' property should be provided having value of the comma separated list of paths to cql files");
+
+        final Collection<Path> directories = Arrays.stream(directoriesProperty.split(","))
```

Comment:
Do we wish to preserve ordering of the directories? If so I'd keep this typed to a List